vector/src/androidTest/java/im/vector/app/ui/robot/ElementRobot.kt

```diff
@@ -88,9 +96,42 @@ class ElementRobot {
     }
 
     fun signout() {
-        OnboardingRobot().signout()
+        clickOn(R.id.groupToolbarAvatarImageView)
+        clickOn(R.id.homeDrawerHeaderSignoutView)
+
+        val hasSentMessages = kotlin.runCatching {
+            waitUntilViewVisible(withId(R.id.exitAnywayButton))
+        }.isSuccess
```

Comment:
That's a clever way to do it, but which will never make the test fail if it should fail. Maybe add a param `expectSignOutWarning: Boolean` to compare with?

Reply:
I originally had the `hasSentMessages` as a parameter but the verification popup and exit flow isn't always consistent 

Adding a warning log sounds like the best of both worlds, will add :+1: